alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -351,6 +360,125 @@ class AlertService(
     }
 
+    suspend fun clearMonitorErrorAlert(monitor: Monitor) {
+        try {
+            val searchRequest = SearchRequest("${monitor.dataSources.alertsIndex}*")
+                .source(
+                    SearchSourceBuilder()
+                        .sort(Alert.START_TIME_FIELD, SortOrder.DESC)
+                        .query(
+                            QueryBuilders.boolQuery()
+                                .must(QueryBuilders.queryStringQuery("${Alert.ALERT_ID_FIELD}:$ERROR_ALERT_ID_PREFIX*"))
+                                .must(QueryBuilders.termQuery(Alert.MONITOR_ID_FIELD, monitor.id))
+                                .must(QueryBuilders.termQuery(Alert.STATE_FIELD, Alert.State.ERROR))
+                        )
+                )
+            val searchResponse: SearchResponse = client.suspendUntil { search(searchRequest, it) }
+
+            if (searchResponse.hits.totalHits.value > 0L) {
+                if (searchResponse.hits.totalHits.value > 1L) {
+                    logger.warn("Found [${searchResponse.hits.totalHits.value}] error alerts for monitor [${monitor.id}] while clearing")
+                }
+                // Deserialize first/latest Alert
+                val hit = searchResponse.hits.hits[0]
+                val xcp = contentParser(hit.sourceRef)
+                val existingErrorAlert = Alert.parse(xcp, hit.id, hit.version)
+
+                val updatedAlert = existingErrorAlert.copy(
+                    endTime = Instant.now()
+                )
+
+                val indexRequest = IndexRequest(monitor.dataSources.alertsIndex)
+                    .routing(monitor.id)
+                    .id(updatedAlert.id)
+                    .source(updatedAlert.toXContentWithUser(XContentFactory.jsonBuilder()))
+                    .opType(DocWriteRequest.OpType.INDEX)
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+
+                val indexResponse: IndexResponse = client.suspendUntil { index(indexRequest, it) }
+                logger.debug("Error alert successfully cleared. End time set to: ${updatedAlert.endTime}")
+            }
+        } catch (e: Exception) {
+            logger.error("Error clearing monitor error alert for monitor [${monitor.id}]: ${ExceptionsHelper.detailedMessage(e)}")
+        }
+    }
+
+    /**
+     * Moves already cleared "error alerts" to history index.
+     * Error Alert is cleared when endTime timestamp is set, on first successful run after failed run
+     * */
+    suspend fun moveClearedErrorAlertsToHistory(alertIndex: String, alertHistoryIndex: String) {
+        try {
+            val searchRequest = SearchRequest(alertIndex)
+                .source(
+                    SearchSourceBuilder()
+                        .query(
+                            QueryBuilders.boolQuery()
+                                .must(QueryBuilders.queryStringQuery("${Alert.ALERT_ID_FIELD}:$ERROR_ALERT_ID_PREFIX*"))
```

Comment:
In general we want to move all completed alerts to history index, so we should then make this generic to look for completed alerts or has end time.